sklearn/ensemble/_hist_gradient_boosting/_predictor.pyx

```diff
@@ -21,46 +21,162 @@ from .common cimport node_struct
 np.import_array()
 
 
+@cython.final
+cdef class TreePredictor:
+    """Tree class used for predictions.
+
+    Parameters
+    ----------
+    nodes_cnt : int
+        Number of nodes in tree.
+    """
+    def __init__(self, int node_cnt):
+        self._nodes.resize(node_cnt)
+
+    cdef node_struct* get(self, int node_idx) nogil:
+        """Get reference to node."""
+        return &self._nodes[node_idx]
+
+    cdef int get_size(self) nogil:
+        """Get number of nodes in tree."""
+        return self._nodes.size()
+
+    @property
+    def nodes(self):
```

Comment:
Why have a private `_nodes` attribute? Should we just have a public `nodes` attribute?